python/ray/train/tests/test_backend.py

```diff
@@ -169,6 +131,32 @@ def train_func():
     assert set(e.finish_training()) == {0, 1}
 
 
+def test_local_world_size(ray_2_node_3_worker):
+    config = TestConfig()
+    with patch.object(WorkerGroup, "add_workers", mock_add_workers):
+        e = BackendExecutor(config, num_workers=3)
+        e.start()
+
+        def train_func():
+            return session.get_local_world_size()
+
+        e.start_training(train_func, dataset_spec=EMPTY_RAY_DATASET_SPEC)
+        assert set(e.finish_training()) == {2, 1}
+
+
+def test_node_ranks(ray_2_node_3_worker):
+    config = TestConfig()
+    with patch.object(WorkerGroup, "add_workers", mock_add_workers):
+        e = BackendExecutor(config, num_workers=3)
+        e.start()
+
+        def train_func():
+            return session.get_node_rank()
+
+        e.start_training(train_func, dataset_spec=EMPTY_RAY_DATASET_SPEC)
+        assert set(e.finish_training()) == {0, 1}
```

Comment:
Why is there only 2 results if num_workers is 3?

Reply:
it's a set

Reply:
why use a set? we're not fully testing the behavior then, right?

Reply:
updated